Use irange in PyTorch

sjoakley opened the pull request:
Summary: Replacing increment iterator loops with ranged loops. It allows loops such as for(int i=0;i<10;i++) to be expressed as for(const auto i : c10::irange(10)). This auto-types the loops and adds const-safety to the iteration variable.

Differential Revision: D34136539

albanD reviewed (approved):
SGTM

albanD commented on torch/csrc/autograd/engine.cpp:
```diff
@@ -603,7 +603,7 @@ void GraphTask::exec_post_processing() {
     // added in between callback calls, so iterators may become invalidated.
     // NOLINTNEXTLINE(modernize-loop-convert)
-    for (size_t i = 0; i < final_callbacks_.size(); ++i) {
+    for (const auto i : c10::irange(final_callbacks_.size())) {
```
Sorry read that one too quickly. As mentioned in the comment above, this should not be changed!

sjoakley committed:
Use irange in PyTorch (#72836)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/72836

Replacing increment iterator loops with ranged loops. It allows loops such as for(int i=0;i<10;i++) to be expressed as for(const auto i : c10::irange(10)). This auto-types the loops and adds const-safety to the iteration variable.

Reviewed By: albanD

Differential Revision: D34136539

fbshipit-source-id: a83a0caf3cb9b3e53a8f29c19ecae05455e4d7e0

albanD reviewed (approved):
Thanks!
